lib/manager/components/version/FeedVersionTabs.js

```diff
@@ -101,21 +105,37 @@ export default class FeedVersionTabs extends Component<Props> {
           unmountOnExit>
           <Tab eventKey={'feed'} title='Summary'>
             <Row>
-              {countFields.map(c => (
-                <Col xs={2} className='text-center' key={c}>
-                  <p
-                    title={`${version.validationSummary[c]}`}
-                    style={{marginBottom: '0px', fontSize: '200%'}}>
-                    {numeral(version.validationSummary[c]).format('0 a')}
-                  </p>
-                  <p style={{marginBottom: '0px'}}>{this.messages(c)}</p>
-                </Col>
-              ))}
+              {countFields.map(c => {
+                const summary = version.validationSummary
+                const diff = comparedVersion ? (summary[c] - comparedVersion.validationSummary[c]) : 0
+
+                return (
+                  <Col xs={2} className='text-center' key={c}>
+                    <div
+                      title={`${summary[c]}`} // FIXME: do we need the title prop?
+                      style={{marginBottom: '0px', fontSize: '200%'}}>
+                      {numeral(summary[c]).format('0a')}
+                      {comparedVersion && comparedVersionIndex && (
+                        <sup style={{width: 0, display: 'inline-block'}}>
+                          <BsLabel
+                            style={{fontSize: '50%', padding: '0 0.2em 0.1em'}}
+                            bsStyle={diff > 0 ? 'success' : (diff < 0 ? 'danger' : 'default')}
+                            title={`Difference compared to Version ${comparedVersionIndex || 'null'}`}>
+                            {diff === 0 ? '=' : ((diff > 0 ? '+' : '') + numeral(diff).format('0a')) /* TODO: refactor */}
```

Comment:
All of these calculations are kind of hard to read when inlined inside the jsx. Can you make them into variables that are calculated before the return statement?

Reply:
I extracted a function in https://github.com/ibi-group/datatools-ui/pull/528/commits/196750f9a02eee12b8bb1f021fce8632be7db654 as this is used in two locations.